### ISSUE TITLE
Bundling image set as href target

### ISSUE THREAD

**Issue by bre7** [parcel-bundler/parcel#1917]
# ❔ Question

Can images that are the target of an anchor link be bundled by Parcel ? 

## 🔦 Context

Since assets are moved to the root of the site by Parcel, I would like to do the same with other images. In this case, full-size images aren't displayed on the website but instead displayed only to users that click on the thumbnail (using Photoswipe).

## 💻 Code Sample

```html
<a href="images/foo.jpeg" itemprop="contentUrl">
    <img src="images/foo_400w.jpeg">
</a>
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.9.7
| Node             | 10.6.0
| npm/Yarn         | 1.9.4
| Operating System | Win 10

**Comment by DeMoorJasper** [comment]
This should already work? Is this inside html or Javascript?

Inside javascript you should use `require(...)` inside html it should already work, you could try with `./images...` to be more explicit but it should work

**Comment by bre7** [comment]
HTML. As is, Parcel will only bundle the 400w image (using parcel-plugin-lazy). But the folder structure of "images/foo.jpeg" will remain unmodified after building.

**Comment by DeMoorJasper** [comment]
Could you try without the plugin?

It's probably an issue with the plugin

**Comment by DeMoorJasper** [comment]
Ow it actually creates the output dir and everything, that's pretty strange

**Comment by bre7** [comment]
Isn't it expected ?  https://github.com/parcel-bundler/parcel/pull/1025 states:

> a hash of bundle content is appended to names of files that aren't entry points (e.g. HTML files, **things linked to a <a href>**, service workers, etc.).

**Comment by DeMoorJasper** [comment]
Yeah it is as `a` tags are handled without hashing for SEO and such.

Not sure how to resolve this tbh.

I originally thought it just didn't copy to the dist and gave a 404

**Comment by bre7** [comment]
I do have a possible solution, with no API breaking changes to Parcel.

By default: "_things linked to a_" aren't hashed
Opt-in: Either flag to enable/disable. Or better yet, add a list of extensions that should be ignored (e.g: jpeg, etc...)

One way to do it would be setting `nonHashableAssetExtensions` to "*" by default. If it's not, use it's values to check if the href includes an asset listed in the attribute or not.

**Comment by DeMoorJasper** [comment]
@bre7 Alright I'll change this to a feature request not sure if this actually should be the case.

**Comment by bre7** [comment]
Follow up: Jasper, I think this is resulting in non-existent entry points.

Minimal example: https://github.com/bre7/parceljs-issue-1917-v2

```bash
parcel build --no-cache --detailed-report 10 --no-source-maps dev/index.html
```

Result:

> √  Built in 1.64s.
>
>dist\images\gallery\3.jpeg    88.05 KB    879ms
dist\images\gallery\2.jpeg    65.82 KB    806ms
dist\images\gallery\1.jpeg    30.05 KB    745ms
dist\images\gallery\4.jpeg    25.75 KB    769ms
dist\index.html                2.11 KB    639ms
dist\images\gallery\1.js        1.1 KB    747ms
dist\images\gallery\2.js        1.1 KB    807ms
dist\images\gallery\3.js        1.1 KB    881ms
dist\images\gallery\4.js        1.1 KB    770ms
Done in 3.00s.

Expected:

> √  Built in 1.64s.
>
>dist\images\gallery\3.jpeg    88.05 KB    879ms
dist\images\gallery\2.jpeg    65.82 KB    806ms
dist\images\gallery\1.jpeg    30.05 KB    745ms
dist\images\gallery\4.jpeg    25.75 KB    769ms
dist\index.html                2.11 KB    639ms
Done in 3.00s.

❌ 1.js / 2.js / 3.js / 4.js don't exist
